components/iam/pkg/oidc/service.go

```diff
@@ -15,59 +15,56 @@ import (
 
 	"github.com/coreos/go-oidc/v3/oidc"
 	"github.com/gitpod-io/gitpod/common-go/log"
-	"github.com/go-chi/chi/v5"
 	"golang.org/x/oauth2"
 )
 
-type OIDCService struct {
-	Handler chi.Router
-
-	configsById      map[string]*OIDCClientConfig
+type Service struct {
+	configsById      map[string]*ClientConfig
 	verifierByIssuer map[string]*oidc.IDTokenVerifier
 	providerByIssuer map[string]*oidc.Provider
 }
 
-type OIDCClientConfig struct {
+type ClientConfig struct {
 	ID           string
 	Issuer       string
 	OAuth2Config *oauth2.Config
 	OIDCConfig   *oidc.Config
 }
 
-type OIDCStartParams struct {
+type StartParams struct {
 	State       string
 	Nonce       string
 	AuthCodeURL string
 }
 
-type OIDCAuthResult struct {
+type AuthResult struct {
 	IDToken *oidc.IDToken
 }
 
-func NewOIDCService() *OIDCService {
-	var s OIDCService
-	s.configsById = make(map[string]*OIDCClientConfig)
-	s.verifierByIssuer = make(map[string]*oidc.IDTokenVerifier)
-	s.providerByIssuer = make(map[string]*oidc.Provider)
-	return &s
+func NewService() *Service {
+	return &Service{
+		configsById:      map[string]*ClientConfig{},
+		verifierByIssuer: map[string]*oidc.IDTokenVerifier{},
+		providerByIssuer: map[string]*oidc.Provider{},
+	}
```

Comment:
For cases where we need to initialize maps, this is the more common pattern. In general, direct field assignment is used sparingly as it easily leads to missing a field which gets added in the future.